### PR TITLE
add channel affinity property to start recording

### PR DESCRIPTION
**Pull request by ikumarapeli-msft**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/communication/azure-communication-callautomation/azure/communication/callautomation/_call_automation_client.py [hunk 1]
```diff
@@ -424,6 +425,12 @@ def start_recording(
          which participant first audio was detected.
          Channel to participant mapping details can be found in the metadata of the recording.
         :paramtype audio_channel_participant_ordering: list[~azure.communication.callautomation.CommunicationIdentifier]
+        :keyword channel_affinity: The channel affinity of call recording
+         When 'recordingChannelType' is set to 'unmixed', if channelAffinity is not specified,
+         'channel' will be automatically assigned.
+         Channel-Participant mapping details can be found in the metadata of the recording.
+         ///.
```
Comment (minwoolee-msft) [inline sdk/communication/azure-communication-callautomation/azure/communication/callautomation/_call_automation_client.py:432]: Question: what is this? \\\.

File: sdk/communication/azure-communication-callautomation/azure/communication/callautomation/_models.py [hunk 2]
```diff
@@ -134,6 +136,41 @@ def _to_generated(self):
         return CallLocator(kind=self.kind,
                            group_call_id=self.id)
 
+class ChannelAffinity(object):
+    """Channel affinity for a participant.
+
+    All required parameters must be populated in order to send to Azure.
+
+    :ivar channel: Channel number to which bitstream from a particular participant will be written.
```
Comment (minwoolee-msft) [inline sdk/communication/azure-communication-callautomation/azure/communication/callautomation/_models.py:144]: nit: in docstring, channel comes first but in actual signature, target_participant comes first in order

File: sdk/communication/azure-communication-callautomation/azure/communication/callautomation/aio/_call_automation_client_async.py [hunk 3]
```diff
@@ -425,6 +426,12 @@ async def start_recording(
          which participant first audio was detected.
          Channel to participant mapping details can be found in the metadata of the recording.
         :paramtype audio_channel_participant_ordering: list[~azure.communication.callautomation.CommunicationIdentifier]
+        :keyword channel_affinity: The channel affinity of call recording
+         When 'recordingChannelType' is set to 'unmixed', if channelAffinity is not specified,
+         'channel' will be automatically assigned.
+         Channel-Participant mapping details can be found in the metadata of the recording.
+         ///.
```
Comment (juntuchen-msft) [inline sdk/communication/azure-communication-callautomation/azure/communication/callautomation/aio/_call_automation_client_async.py:433]: Same as Min's comment above

Comment (ikumarapeli-msft) [reply sdk/communication/azure-communication-callautomation/azure/communication/callautomation/aio/_call_automation_client_async.py:433]: removed additional line

File: sdk/communication/azure-communication-callautomation/azure/communication/callautomation/_call_automation_client.py (hunk 1 above)
Comment (ikumarapeli-msft) [reply sdk/communication/azure-communication-callautomation/azure/communication/callautomation/_call_automation_client.py]: removed additional line

File: sdk/communication/azure-communication-callautomation/azure/communication/callautomation/_models.py (hunk 2 above)
Comment (ikumarapeli-msft) [reply sdk/communication/azure-communication-callautomation/azure/communication/callautomation/_models.py]: reordered

**Review by minwoolee-msft — APPROVED** [review]

File: sdk/communication/azure-communication-callautomation/azure/communication/callautomation/aio/_call_automation_client_async.py [hunk 4]
```diff
@@ -425,6 +426,11 @@ async def start_recording(
          which participant first audio was detected.
          Channel to participant mapping details can be found in the metadata of the recording.
         :paramtype audio_channel_participant_ordering: list[~azure.communication.callautomation.CommunicationIdentifier]
+        :keyword channel_affinity: The channel affinity of call recording
```
[the remark this answers is not in the document]
Comment (ikumarapeli-msft) [reply sdk/communication/azure-communication-callautomation/azure/communication/callautomation/aio/_call_automation_client_async.py:432]: updated order

**Pull request merged**